src/prefect/new_flow_engine.py

```diff
@@ -179,19 +176,8 @@ async def create_subflow_task_run(
         dummy_task = Task(
             name=self.flow.name, fn=self.flow.fn, version=self.flow.version
         )
-        task_inputs = {
-            k: await collect_task_run_inputs(v) for k, v in self.parameters.items()
-        }
-        parent_task_run = await client.create_task_run(
-            task=dummy_task,
-            flow_run_id=(
-                context.flow_run.id if getattr(context, "flow_run", None) else None
-            ),
-            dynamic_key=_dynamic_key_for_task_run(context, dummy_task),
-            task_inputs=task_inputs,
-            state=Pending(),
-        )
-        return parent_task_run
+        task_engine = TaskRunEngine(task=dummy_task, parameters=self.parameters)
+        return await task_engine.create_task_run(client)
```

Comment:
Absolutely, that feels clean to me. And avoids the engine object feeling overloaded too. The reason I consolidated here was just to ensure that the "dummy" subflow task was being created with exactly the same logic we used for all other tasks, so this sounds very in line. Happy to do that here or support the change in #13153, whatever order of operations plays out.

Reply:
Quick update: I moved this logic to use the new method but it doesn't include the __parent__ key and other logic still on the task run engine, so some things failed. Rather than mirror all of that into `Task.create_run()` in this PR, I opted to leave as is and if/when you use migrate `TaskRunEngine.create_task_run` over to the new method, this will go with it smoothly.